### PR TITLE
add option allowing users to use cmake toolchain

### PR DESCRIPTION
**Pull request by yunqu**
Removed pynqlinux from native build, since pynq is going through the non-native build for now. In the future we probably need better ways to decide which kind of platform users are building upon.

Add extra toolchain option, so users can provide an external toolchain file (toolchain.cmake). This option, if ignored (by default), will not take any effect.

### CODE REVIEW

File: build/build.sh [hunk 1]
```diff
@@ -180,8 +188,27 @@ if [[ $dbg == 1 ]]; then
...
+		echo "	-DCMAKE_BUILD_TYPE=Debug"
+		echo "	-DCMAKE_EXPORT_COMPILE_COMMANDS=ON"
+		echo "	-DCMAKE_TOOLCHAIN_FILE=$toolchain"
+		echo "	../../src"
+		time $CMAKE -DRDI_CCACHE=$ccache \
+			-DCMAKE_BUILD_TYPE=Debug \
+			-DCMAKE_EXPORT_COMPILE_COMMANDS=ON \
+			-DCMAKE_TOOLCHAIN_FILE=$toolchain \
+			../../src
+	fi
```
Comment (stsoe) [inline build/build.sh]: I believe you can simply always pass -DCMAKE_TOOLCHAIN_FILE=$toolchain without checking if the variable was specified.   If it is empty -DCMAKE_TOOLCHAIN_FILE will simply be ignored.  Please check.

I'd prefer to have this be one line commands as before.

File: build/build.sh [hunk 2]
```diff
@@ -193,8 +220,27 @@ if [[ $opt == 1 ]]; then
...
+		echo "	-DCMAKE_BUILD_TYPE=Release"
+		echo "	-DCMAKE_EXPORT_COMPILE_COMMANDS=ON"
+		echo "	-DCMAKE_TOOLCHAIN_FILE=$toolchain"
+		echo "../../src"
+        time $CMAKE -DRDI_CCACHE=$ccache \
+			-DCMAKE_BUILD_TYPE=Release \
+			-DCMAKE_EXPORT_COMPILE_COMMANDS=ON \
+			-DCMAKE_TOOLCHAIN_FILE=$toolchain \
+			../../src
+	fi
```
Comment (stsoe) [inline build/build.sh]: Same as earlier comment

File: build/build.sh (hunk 1 above)
Comment (yunqu) [reply build/build.sh]: @stsoe The purpose of using multiple lines is to make the code rendering on github as well as vim editors much better - and if you have a very long single line, then even small changes would be very difficult to track.

Comment (stsoe) [reply build/build.sh]: @yunqu  I hear you, but in this case I thought it made it less readable.  Thanks for fixing, it looks good now (at least to me) :-)

**Pull request merged**